src/storage.rs

```diff
@@ -183,3 +185,23 @@ pub trait HashInterface {
         data_hash: Vec<u8>,
     ) -> Result<types::HashTable, ContainerError<Self::Error>>;
 }
+
+///
+/// Fingerprint:
+///
+/// Interface providing functional to interface with the fingerprint table in database
+#[async_trait::async_trait]
+pub trait FingerprintInterface {
+    type Error;
+
+    async fn find_by_card_hash(
+        &self,
+        card_hash: &[u8],
```

Comment:
Can we have this as secret?

Reply:
Hey this still needs to be a secret, maybe something like `&masking::Secret<[u8]>` or `masking::Secret<&[u8]>`